sup3r/bias/qdm.py

```diff
@@ -213,10 +217,42 @@ def _init_out(self):
                 f'bias_fut_{self.bias_feature}_params',
                 f'base_{self.base_dset}_params',
                 ]
-        shape = (*self.bias_gid_raster.shape, self.n_quantiles)
+        shape = (*self.bias_gid_raster.shape, self.NT, self.n_quantiles)
         arr = np.full(shape, np.nan, np.float32)
         self.out = {k: arr.copy() for k in keys}
 
+        self.time_window_center = self._window_center(self.NT)
+
+    @staticmethod
+    def _window_center(ntimes: int):
+        """A sequence of equally spaced `ntimes` day of year along a year
+
+        This is used to identify the center of moving windows to apply filters
+        and masks. For instance, if ntimes equal to 12, it would return
+        approximately the months' center time.
+
+        This is conveniently shifted one half time interval so that December
+        31st would be closest to the last interval of the year instead of the
+        first.
+
+        Leap years are neglected here.
+
+        Parameters
+        ----------
+        ntimes : int
+            Number of intervals in one year.
+
+        Returns
+        -------
+        np.ndarray :
+            Sequence of center points dividing a standard year in `ntimes`
+            intervals.
+        """
+        assert ntimes > 0, "Requires a positive number of intervals"
+
+        dt = 365 / ntimes
+        return np.arange(dt / 2, 366, dt)
```

Comment:
That's a good point. It should not fail. This step only gives the window center, so it should be independent of the time interval of `base_data`. We can choose a time window centered anywhere.

When gets to the point to filter the data within a window, `base_ti` is a `pd.DatetimeIndex`, so extracting day of the year should be safe. That would lead to a rounding error since I'm using just `day_of_year` without including the fraction from the hours/minutes. 

I just added a new method `doy_fractional` that we can use or remove it if you're concerned with that rounding error.

About the second part.
If we alternate a fraction of 365 or 366 days, that would result in a slightly variable window. For instance a 1/12 window size would be slightly different for 365 and 366.
 
A more precise solution would be to count continuous time from the start of the time series, divide that running time by 365.25, remove the integer, and use the fraction of an actual year. This would avoid the shift from jumping one year every 4 years. I considered that negligible before. One issue of this approach would be processing `no_leap` type of simulations.

I think that the error should be minimal. The time window is applied everywhere, but any window extending through the change of year would have a few data more to average, slightly biasing the average on that direction.

I suggest keep it as it is, but I'm glad to change if you have a strong opinion about it.

Reply:
Can you push? I don't see new commits that you mentioned. 

Okay i think I see your point about how this will actually totally work with hourly data because DOY is being used in the indexing. Maybe note in the _window_center docstring that the output array indexes the day of year? Looks good to me! I'm not worried about fractional days for leap years.

Reply:
Sure. I'll add that extra info in the documentation in the PresRat branch (after merging with this) so we don't hold this one here any longer.